src/elements/element.rectangle.js

```diff
@@ -54,12 +56,104 @@ function getBarBounds(bar) {
 	};
 }
 
+function drawLines(ctx, lines) {
+	var i = 0;
+	var ilen = lines.length;
+	var line, prevLine;
+
+	for (; i < ilen; ++i) {
+		line = lines[i];
+		if (i === 0 || prevLine.w !== line.w) {
+			if (i > 0) {
+				ctx.stroke();
+			}
+			ctx.beginPath();
+			ctx.lineWidth = line.w;
+		}
+		if (i === 0 || prevLine.x2 !== line.x1 || prevLine.y2 !== line.y1) {
+			ctx.moveTo(line.x1, line.y1);
+		}
+		ctx.lineTo(line.x2, line.y2);
+		prevLine = line;
+	}
+	if (ilen) {
+		ctx.stroke();
+	}
+}
+
+// eslint-disable-next-line complexity
+function buildBorderLines(rect, width, offset) {
+	var lines = [];
+	var halfOffsetLeft = offset.left / 2;
+	var halfOffsetRight = offset.right / 2;
+	var halfOffsetTop = offset.top / 2;
+	var halfOffsetBottom = offset.bottom / 2;
+
+	if (width.bottom) {
+		lines.push({
+			w: width.bottom,
+			x1: rect.right,
+			y1: rect.bottom + halfOffsetBottom,
+			x2: rect.left + (width.bottom === width.left ? halfOffsetLeft : offset.left),
+			y2: rect.bottom + halfOffsetBottom
+		});
+	}
+	if (width.left) {
+		lines.push({
+			w: width.left,
+			x1: rect.left + halfOffsetLeft,
+			y1: rect.bottom + (width.bottom === width.left ? halfOffsetBottom : 0),
+			x2: rect.left + halfOffsetLeft,
+			y2: rect.top + (width.left === width.top ? halfOffsetTop : offset.top)
+		});
+	}
+	if (width.top) {
+		lines.push({
+			w: width.top,
+			x1: rect.left + (width.left === width.top ? halfOffsetLeft : 0),
+			y1: rect.top + halfOffsetTop,
+			x2: rect.right + (width.top === width.right ? halfOffsetRight : offset.right),
+			y2: rect.top + halfOffsetTop
+		});
+	}
+	if (width.right) {
+		lines.push({
+			w: width.right,
+			x1: rect.right + halfOffsetRight,
+			y1: rect.top + (width.top === width.right ? halfOffsetTop : 0),
+			x2: rect.right + halfOffsetRight,
+			y2: rect.bottom + offset.bottom
+		});
+	}
+	return lines;
+}
+
+function parseBorderWidth(borderWidth, borderSkipped, maxWidth, maxHeight) {
+	if (helpers.isObject(borderWidth)) {
+		return {
+			bottom: Math.min(borderWidth.bottom || 0, maxHeight),
+			left: Math.min(borderWidth.left || 0, maxWidth),
+			top: Math.min(borderWidth.top || 0, maxHeight),
+			right: Math.min(borderWidth.right || 0, maxWidth)
+		};
+	}
+
+	maxWidth = Math.min(borderWidth, maxWidth);
+	maxHeight = Math.min(borderWidth, maxHeight);
+	return {
+		bottom: borderSkipped === 'bottom' ? 0 : maxHeight,
+		left: borderSkipped === 'left' ? 0 : maxWidth,
+		top: borderSkipped === 'top' ? 0 : maxHeight,
+		right: borderSkipped === 'right' ? 0 : maxWidth
+	};
+}
```

Comment:
I think `borderSkipped` (even if deprecated) should be applied on top of `borderWidth`, whatever the format of this last, else it's not consistent and a bit confusing. I would rewrite this method as:

```javascript
function parseBorderWidth(value, skipped, maxWidth, maxHeight) {
		var t, r, b, l;

		if (helpers.isObject(value)) {
			t = +value.top || 0;
			r = +value.right || 0;
			b = +value.bottom || 0;
			l = +value.left || 0;
		} else {
			t = r = b = l = +value || 0;
		}

		return {
			top: Math.min(maxHeight, skipped=== 'top' ? 0 : t),
			right: Math.min(maxWidth, skipped === 'right' ? 0 : r),
			bottom: Math.min(maxHeight, skipped=== 'bottom' ? 0 : b),
			left: Math.min(maxWidth, skipped=== 'left' ? 0 : l)
		};
}
```

Reply:
The problem with this is, we provide a default for `borderSkipped`.
So adding `borderWidth: {bottom: 10}` on a vertical bar chart would have no effect if `borderSkipped` was not also defined.
Is that the way we want it to work?

Reply:
I think it's more consistent (and easier to maintain) if `borderSkipped` applies independently to the format of `borderWidth`. So I would rather add a note in the docs to explain that in order to make this case working, `borderSkipped` need to be set to `false`.

Reply:
I like @simonbrunel's suggestion